src/commands/threat-feed/get-threat-feed.ts

```diff
@@ -22,87 +28,169 @@ type ThreatResult = {
   apiToken,
   direction,
+  ecoSystem,
   filter,
-  outputJson,
+  outputKind,
   page,
   perPage
 }: {
   apiToken: string
-  outputJson: boolean
-  perPage: number
-  page: string
   direction: string
+  ecoSystem: string
   filter: string
+  outputKind: 'json' | 'markdown' | 'print'
+  page: string
+  perPage: number
 }): Promise<void> {
   // Lazily access constants.spinner.
   const { spinner } = constants
 
-  spinner.start('Looking up the threat feed')
+  const queryParams = new URLSearchParams([
+    ['direction', direction],
+    ['ecosystem', ecoSystem],
+    ['filter', filter],
+    ['page', page],
+    ['per_page', String(perPage)]
+  ])
 
-  const formattedQueryParams = formatQueryParams({
-    per_page: perPage,
-    page,
-    direction,
-    filter
-  }).join('&')
-  const response = await queryAPI(
-    `threat-feed?${formattedQueryParams}`,
-    apiToken
-  )
+  spinner.start('Fetching Threat Feed data...')
+
+  const response = await queryAPI(`threat-feed?${queryParams}`, apiToken)
   const data = <{ results: ThreatResult[]; nextPage: string }>(
     await response.json()
   )
 
-  spinner.stop()
+  spinner.stop('Threat feed data fetched')
 
-  if (outputJson) {
+  if (outputKind === 'json') {
     logger.log(data)
     return
   }
 
-  const screen = new ScreenWidget()
+  const screen: Widgets.Screen = new ScreenWidget()
 
-  const table = new TableWidget({
+  const table: any = new TableWidget({
     keys: 'true',
     fg: 'white',
     selectedFg: 'white',
     selectedBg: 'magenta',
     interactive: 'true',
     label: 'Threat feed',
     width: '100%',
-    height: '100%',
+    height: '70%', // Changed from 100% to 70%
     border: {
       type: 'line',
       fg: 'cyan'
     },
-    columnSpacing: 3, //in chars
-    columnWidth: [9, 30, 10, 17, 13, 100] /*in chars*/
+    columnWidth: [10, 30, 20, 18, 15, 200],
+    // TODO: the truncation doesn't seem to work too well yet but when we add
+    //       `pad` alignment fails, when we extend columnSpacing alignment fails
+    columnSpacing: 1,
+    truncate: '_'
+  })
+
+  // Create details box at the bottom
+  const detailsBox: Widgets.BoxElement = new BoxWidget({
+    bottom: 0,
+    height: '30%',
+    width: '100%',
+    border: {
+      type: 'line',
+      fg: 'cyan'
+    },
+    label: 'Details',
+    content:
+      'Use arrow keys to navigate. Press Enter to select a threat. Press q to exit.',
+    style: {
+      fg: 'white'
+    }
   })
 
   // allow control the table with the keyboard
   table.focus()
 
   screen.append(table)
+  screen.append(detailsBox)
 
   const formattedOutput = formatResults(data.results)
+  const descriptions = data.results.map(d => d.description)
 
   table.setData({
     headers: [
-      'Ecosystem',
-      'Name',
-      'Version',
-      'Threat type',
-      'Detected at',
-      'Details'
+      ' Ecosystem',
+      ' Name',
+      '  Version',
+      '  Threat type',
+      '  Detected at',
+      ' Details'
```

Comment:
☝️ Can you add a comment about the spacing hacks

Reply:
Sure, copied the comment I added below here too